mountpoint-s3/src/upload.rs

```diff
@@ -125,7 +132,7 @@ impl<Client: ObjectClient> UploadRequest<Client> {
         let size = self.size();
         let checksum = self.hasher.finalize();
         self.request
-            .review_and_complete(move |review| verify_checksums(review, size, checksum))
+            .review_and_complete(move |review| verify_checksums(review, size, checksum, self.set_checksum))
```

Comment:
You could skip the review completely and just call `PutObjectRequest::complete()` here.

Reply:
Alternatively, you could still use a review to verify just the size. In either cases, you do not want to calculate the checksum if not using it.